testing/execution_engine_integration/src/geth.rs

```diff
@@ -23,20 +22,17 @@ pub fn build(execution_clients_dir: &Path) {
 
     if !repo_dir.exists() {
         // Clone the repo
-        assert!(build_utils::clone_repo(
-            execution_clients_dir,
-            GETH_REPO_URL
-        ));
+        build_utils::clone_repo(execution_clients_dir, GETH_REPO_URL).unwrap();
     }
 
-    // Checkout the correct branch
-    assert!(build_utils::checkout_branch(&repo_dir, GETH_BRANCH));
-
-    // Update the branch
-    assert!(build_utils::update_branch(&repo_dir, GETH_BRANCH));
+    // Get the latest tag on the branch
+    let last_release = build_utils::get_latest_release(&repo_dir).unwrap();
```

Comment:
I think this is slightly awkward in the case where `repo_dir` already existed and was checked out to a prior release. The `git describe` will describe the `HEAD` so it will just return the same tag back, and the repo will be stuck until the user manually intervenes. I don't think this is a problem for CI, however.

I guess the `update_branch` function is meant to address this, but it's unused in the current code. Alternatively we could do `git describe origin/{branch_name}` (after the `git fetch`), which avoids the problems of statefullness.

I'm inclined to merge this PR now as-is and we can come back for these changes if others agree?

Reply:
So I tried the describe approach and didn't work. I'll just do checkout branch, pull, fetch tags and then describe. What do you think?

Reply:
I think that's fine! I'm surprised the describe approach doesn't work though

Reply:
oh I think it might be because I testing against my lighthouse repo and I've never updated my remote master, let me try again

Reply:
yep it works